File: Practica3_121/Python/Pila.py
class Pila:
    def __init__(self):
        self.arreglo = []
        
    def push(self,e):
        self.arreglo.append(e)
        
    def pop(self):
        e = -1
        if not self.isEmpty():
            e = self.arreglo.pop()
        else:
            print("Pila vacía.")
      
        return e

    def peek(self):
        e = -1
        if not self.isEmpty():
            n = len(self.arreglo)
            e = self.arreglo[n-1]
        else:
            print("Pila vacía.")

        return e
    
    def isEmpty(self):
        return len(self.arreglo) == 0

pila = Pila()

pila.push(100)
pila.push(200)
pila.push(300)
pila.push(400)
pila.push(500)

print("Elemento en la cima:", pila.peek()) 

print("Elemento removido:", pila.pop())  
print("Elemento removido:", pila.pop())  
print("Elemento removido:", pila.pop())  
print("Elemento removido:", pila.pop())  
print("Elemento removido:", pila.pop())  

print("Elemento removido:", pila.pop())  

File: Practica3_121/Java/Pila.java
public class Pila {
	// atributos
	private int n; // tamaño de la pila
	private long[] arreglo;
	private int top; // cima o parte superior de la pila
	
	// métodos
	public Pila(int n) { // constructor
		this.n = n; // asigna el tamaño del arreglo
		arreglo = new long[n]; // crea arreglo
		top = -1; // todavia no tiene elementos
	}
	public void push(long e) { // inserta un elemento en la cima de la pila
		if (!isFull()) {
			top++;            // incrementa top
			arreglo[top] = e; // inserta elemento
		} else
			System.out.println("Pila llena.");
	}
	public long pop() { // elimina un elemento de la cima de la pila
		long e = Long.MIN_VALUE;
		if (!isEmpty()){
			e = arreglo[top]; // retorna el elemento
			top--;            // decrementa top
		} else
			System.out.println("Pila vacía.");
		return e;
	}
	public long peek() { // obtiene el elemento de la cima de la pila
		long e = Long.MIN_VALUE;
		if (!isEmpty()){
			e = arreglo[top]; // retorna el elemento
		} else
			System.out.println("Pila vacía.");
		return e;
	}
	public boolean isEmpty() { // verdad si la pila está vacía
		return (top == -1);
	}
	public boolean isFull() { // verdad si la pila está llena
		return (top == this.n - 1);
	}
}
